### PR TITLE
[Feature] Management cluster registration mock tests

### PR DESCRIPTION
**Pull request by ekoop-terasky**
1. **What this PR does / why we need it**: Management cluster registration mock tests

2. **Which issue(s) this PR fixes**

3. **Additional information**
Related task: [TMC-34852](https://jira.eng.vmware.com/browse/TMC-34852)

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by vrabbi — APPROVED** [review]

File: internal/resources/managementcluster/management_cluster_registration_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	httpmock.RegisterResponder("DELETE", deleteManagementClusterRegistrationResourceEndpoint, changeStateResponder(
+		func() {
+			httpmock.RegisterResponder("GET", readManagementClusterRegistrationEndpoint,
+				httpmock.NewStringResponder(http.StatusBadRequest, "Not found"))
+		},
+		http.StatusOK,
+		nil))
+}
+
+// nolint: unparam
```
Comment (shobha2626) [inline internal/resources/managementcluster/management_cluster_registration_mock_test.go:77]: nit: any reason we have the lint disabled?

File: internal/resources/managementcluster/management_cluster_registration_test.go [hunk 2]
```diff
@@ -31,12 +33,40 @@ func TestAcceptanceForManagementClusterRegistrationResource(t *testing.T) {
...
 
-	tkgsSimpleName := acctest.RandomWithPrefix("a-tf-tkgs-simple-test")
-	tkgmSimpleName := acctest.RandomWithPrefix("a-tf-tkgm-simple-test")
+	tkgsSimpleName := acctest.RandomWithPrefix("a-tf-tkgs-test")
+	tkgmSimpleName := acctest.RandomWithPrefix("a-tf-tkgm-test")
 
 	tkgmKubeconfigFilePathName := acctest.RandomWithPrefix("a-tf-tkgm-kubeconfig-filepath-test")
 
 	kubeconfigPath := os.Getenv("KUBECONFIG")
+	_, enablePolicyEnvTest := os.LookupEnv("ENABLE_POLICY_ENV_TEST")
```
Comment (shobha2626) [inline internal/resources/managementcluster/management_cluster_registration_test.go:42]: POLICY_ENV_TEST?

**Comment by shobha2626** [comment]
Please attach a screenshot of successful mock test run

File: internal/resources/managementcluster/management_cluster_registration_test.go (hunk 2 above)
Comment (ekoop-terasky) [reply internal/resources/managementcluster/management_cluster_registration_test.go]: I renamed it to **ENABLE_MANAGEMENT_CLUSTER_ENV_TEST** to have same approach as other mock tests flags.

**Comment by ekoop-terasky** [comment]
> Please attach a screenshot of successful mock test run

Original:
original

Commented out kubeconfig acceptance test case to have explicit green result:
commented_out

File: internal/resources/managementcluster/management_cluster_registration_mock_test.go (hunk 1 above)
Comment (ekoop-terasky) [reply internal/resources/managementcluster/management_cluster_registration_mock_test.go]: Probably bad leftover. Removed this comment.

**Pull request merged**